drivers/dht/dht.c

```diff
@@ -43,13 +43,32 @@
 /* If an expected pulse is not detected within 1000µs, something is wrong */
 #define TIMEOUT                     (1000U)
 /* The DHT sensor cannot measure more than once a second */
-#define DATA_HOLD_TIME              (1000U * US_PER_MS)
+#define DATA_HOLD_TIME              (US_PER_SEC)
 
 static inline void _reset(dht_t *dev)
 {
     gpio_init(dev->params.pin, GPIO_OUT);
     gpio_set(dev->params.pin);
+}
 
+/**
+ * @brief   Wait until the pin @p pin has level @p expect
+ *
+ * @param   pin     GPIO pin to wait for
+ * @param   expect  Wait until @p pin has this logic level
+ * @param   timeout Timeout in µs
+ *
+ * @retval  0       Success
+ * @retval  -1      Timeout occurred before level was reached
+ */
+static inline int _wait_for_level(gpio_t pin, int expect, unsigned timeout)
```

Comment:
again: I'm unsure if `unsigned timeout` is large enough on all platforms to hold `1.000.000` microseconds, i.e. verify with arduino-mega2560 for instance.

Reply:
on that arduino `int = unsigned = 16Bits`? Which would map 1sec to 65000 microseconds, IIRC?

Reply:
better use `int32_t` to be safe and also avoid underflow (which you correctly pointed out).

Reply:
`unsinged` is guaranteed to be 16 bit by the C standard, so the timeout of 1000 fits quite fine. Also, `unsigned` makes sure that negative timeouts cannot be used, which makes sense. The loop is written to never count below zero.

On an AVR an `unsigned` will be faster than `int32_t`.

Reply:
oops my bad, thought timeout is 1 second and not 1000 us, sry - leave as is with `unsigned` then.